concordium-contracts-common/src/schema.rs

```diff
@@ -1213,6 +1411,24 @@ mod impls {
                     let fields = fields_ty.to_json(source)?;
                     Ok(json!({ name: fields }))
                 }
+                Type::EnumEvent(variants) => {
+                    let enum_index = if variants.len() <= 256 {
+                        let idx = u8::deserial(source)?;
+                        // Find the enum index which is associated to the event tag.
+                        variants
+                            .iter()
+                            .position(|item| item.2 == idx)
+                            .ok_or_else(ParseError::default)?
+                    } else {
+                        // Event schemas are not supported for more than 256 event variants.
+                        return Err(ParseError::default());
+                    };
+
+                    let (name, fields_ty, tag) =
+                        variants.get(enum_index).ok_or_else(ParseError::default)?;
```

Comment:
You could use [`find`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find) here, instead of first finding the index to lookup with.
Something like this:
```suggestion
                        variants.find(|item| item.2 == idx).ok_or_else(ParseError::default)?;
```